File: Game_Of_Life_Python/Organisms/Animals/sheep.py
from Organisms.organism import OrganismInitialData
from Organisms.animal import Animal


class Sheep(Animal):

    static_counter = 0

    def __init__(self, name, row, column, given_strength = -1, given_initiative = -1):
        strength = OrganismInitialData[name]["strength"]
        initiative = OrganismInitialData[name]["initiative"]
        character = OrganismInitialData[name]["character"]
        if given_strength != -1: strength = given_strength
        if given_initiative != -1: initiative = given_initiative
        super().__init__(strength, initiative, name, character, row, column, "sheep.png")
        Sheep.static_counter += 1
        self.type = Sheep

    def action(self, grid_board):
        self.default_action_animal(grid_board)

    def collision(self, grid_board, organisms, current_index):
        return self.default_collision_animal(grid_board, organisms, current_index)


    def get_static_counter(self):
        return Sheep.static_counter

File: GameOfLife_Cpp/source/src/sheep.cpp
#include "sheep.hpp"

int Sheep::SHEEP_STATIC_COUNTER = 0;

Sheep::Sheep(int row, int column)
    : Animal(4, 4, 0, "sheep", 's', row, column, "sheep.png", OrganismType::SHEEP)
{
    SHEEP_STATIC_COUNTER++;  
}

Sheep::Sheep(int row, int column, int strength, int initiative, int age)
    : Animal(strength, initiative, age, "sheep", 's', row, column, "sheep.png", OrganismType::SHEEP)
{
    SHEEP_STATIC_COUNTER++;  
}

Sheep::~Sheep()
{
    SHEEP_STATIC_COUNTER--;
}

ActionResult Sheep::action(std::vector<std::vector<char>>& grid_board)
{
    return Animal::default_action_animal(grid_board);
}

CollisionResult Sheep::collision(std::vector<std::vector<char>>& grid_board, std::vector<Organism*>& organisms, int current_index)
{
    return this->default_collision_animal(grid_board, organisms, current_index);
}

int Sheep::get_static_counter()
{
    return SHEEP_STATIC_COUNTER;
}